minor: fix example in docstring, use np.ceil instead of math.ceil

sappelhoff opened the pull request:
The example was still in Matlab syntax. I fixed it to work in Python.

Furthermore, the module imported math for its `ceil` function and numpy ... I dropped math and use `np.ceil` instead.

cbrnr commented on mne/preprocessing/peak_finder.py:
```diff
@@ -97,7 +96,7 @@ def peak_finder(x0, thresh=None, extrema=1, verbose=None):
 
         # Preallocate max number of maxima
-        maxPeaks = int(ceil(length / 2.0))
+        maxPeaks = int(np.ceil(length / 2.0))
```
No need to convert to `int` since `maxPeaks` is only used in `np.zeros`, which works with integer floats as well.

larsoner replied:
Unfortunately not:
```
>>> import numpy as np
>>> np.zeros(1.)
Traceback (most recent call last):
  File "<ipython-input-2-7226ec7f6b9a>", line 1, in <module>
    np.zeros(1.)

TypeError: 'float' object cannot be interpreted as an integer
```

See also

https://travis-ci.org/mne-tools/mne-python/jobs/370006021#L3548

I would do `max_peaks = (length - 1) // 2 + 1` to avoid `float` entirely.

cbrnr replied:
Sorry, my bad. Yes, this saves converting back and forth between data types, but at the cost of readability. You could also write `-(-length // 2)`, which I like a bit better. Otherwise, maybe let's just keep using `int(np.ceil(length / 2.0))`.

sappelhoff replied:
I think i'd keep it with `int(np.ceil(length / 2.0))` for readability. Unfortunately, the doctests keep failing ... any of you have an idea what's the issue?

cbrnr commented:
Apart from my minor comment, +1 for merge.

cbrnr commented on mne/preprocessing/peak_finder.py:
```diff
@@ -38,10 +37,10 @@ def peak_finder(x0, thresh=None, extrema=1, verbose=None):
-    peak_finder(x)
+    t = np.arange(0, 3, 0.01)
+    x = np.sin(np.pi*t) - np.sin(0.5*np.pi*t)
+    peak_locs, peak_mags = mne.preprocessing.peak_finder.peak_finder(x)
```
Do we really want an example that requires this function from MNE? If so, you should `import mne` in this example.

cbrnr commented:
Thanks @sappelhoff! Regarding the import in the example, could anyone else confirm that we want to have it (e.g. for doctest) or that we don't need it?

larsoner commented on mne/preprocessing/peak_finder.py:
```diff
@@ -38,10 +37,13 @@ def peak_finder(x0, thresh=None, extrema=1, verbose=None):
 
     Example
```
Should be called `Examples`

https://numpydoc.readthedocs.io/en/latest/format.html#sections

sappelhoff replied:
But didn't they just release 1.0.0? How long until 1.1.0 will be out?

larsoner replied:
It is already RC

https://mail.python.org/pipermail/scipy-dev/2018-April/022775.html

larsoner commented on mne/preprocessing/peak_finder.py:
```diff
@@ -38,10 +37,13 @@ def peak_finder(x0, thresh=None, extrema=1, verbose=None):
-    x = 12*sin(10*2*pi*t)-3*sin(.1*2*pi*t)+randn(1,numel(t));
-    x(1250:1255) = max(x);
-    peak_finder(x)
+    import numpy as np
```
I think these need to be preceded by `>>>` to actually be used.

sappelhoff commented:
yup, just removed it (again) ... to be honest, I just added it in an attempt to fix the

```
Expected:
    array([ 36, 260], dtype=int64)
Got:
    array([ 36, 260])
```

but apparently it didn't do anything.

cbrnr commented:
Weird. It seems like output formatting differs between platforms/versions/etc., but now only one Travis job fails:

```
Expected:
    array([0.36900026, 1.76007351])
Got:
    array([ 0.36900026,  1.76007351])
```

One possible hack would be to convert the array to a list:

```
>>> peak_mags.tolist()
[0.36900026, 1.76007351]
```

Ugly, but it works.

cbrnr commented:
What about dropping the doctest and reverting it to a normal example? Now there are differences in displayed digits after the comma...

sappelhoff commented:
let's try this one last time with the expected number of floating points ... if that doesn't work, we can remove the doctest and I'll check how to format the example as a non-doctest example ...

cbrnr commented:
I don't know, comparing floats is always not a good idea. Usually, we would use `np.allclose` to compare two arrays with a certain tolerance.

larsoner commented:
You can also try using `# doctest: +NORMALIZE_WHITESPACE` to take care of the whitespace problem when printing `ndarray`, it might work. If that didn't, you could do `# doctest: +SKIP` to avoid running the line at all. But it looks like whatever you did might already make it pass.

sappelhoff commented:
okay, they passed ... let's break it again with trying `# doctest: + NORMALIZE_WHITESPACE`

sappelhoff commented:
mmmh broken again. Did I use the NORMALIZE_WHITESPACE incorrectly? Otherwise, I'd try the +SKIP next.

cbrnr commented:
The problem is that you don't have any whitespace in the expected output, so this parameter does not work because it matches any sequence of whitespace in the expected output to any sequence of whitespace in the actual output (https://docs.python.org/3/library/doctest.html#doctest.NORMALIZE_WHITESPACE). There is not whitespace in front of the first element in the expected output, so this can't be matched. Also, you have the `dtype` issue on AppVeyor again.

You can either revert to the previous commit and hope that this never breaks, or just skip the doctest business completely here.

sappelhoff commented:
So with this extensive skipping, it's working. It looks bad but at least there is an example.

cbrnr commented:
Yes, it's not pretty. You could also remove these four lines that you skip, because we don't have to show expected results if they are not tested. If you want to leave it in, I'm fine with that too. Either way, +1 for merge.
